### PR TITLE
Fix for CI: adding chart museum ingress

### PR DESCRIPTION
**Pull request by castelblanque**
### Description of the change

CI E2E runs are constantly failing due to problems with the `port-forward` done in the chart-museum part.
Whenever a chart is needed to be pushed to chart-museum, there is a port-forward done to the service.

That solution is weak and error prone, this PR removes that and adds an ingress setup for Chart museum so that there is no port-forward used anymore.
Ingress is only used from the scripts to push charts. From the E2E tests, the same internal service locator is used (`http://chartmuseum.${CHARTMUSEUM_NS}.svc.cluster.local:8080/`).

### Benefits

It should bring more reliability.

### Possible drawbacks

N/A

### Applicable issues

N/A

### CODE REVIEW

File: script/chart-museum.sh [hunk 1]
```diff
@@ -107,9 +109,6 @@ spec:
         pathType: ImplementationSpecific
 EOF
 
-  # Add Ingress IP to hosts file
-  sudo echo "${DEX_IP}  ${CHARTMUSEUM_HOSTNAME}" >> /etc/hosts
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply script/chart-museum.sh]: Not really. That line was a test I did, but I didn't remember that the whole script must be run with `sudo`, which is not a good option. Finally went with sending the `Host` header and going to the IP directly.
However, it still fails and I can't replicate it locally. Whenever I have time I'll continue debugging in CircleCI directly.

[the remark this answers is not in the document]
Comment (castelblanque) [reply script/chart-museum.sh]: Feel free to jump in if you see the cause of the current error.

File: script/chart-museum.sh [hunk 2]
```diff
@@ -88,17 +79,47 @@ installChartMuseum() {
...
+  echo "Installing Ingress for ChartMuseum with access through host ${CHARTMUSEUM_HOSTNAME}"
+  kubectl create -n $CHARTMUSEUM_NS -f - -o yaml << EOF
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    nginx.ingress.kubernetes.io/connection-proxy-header: keep-alive
+    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
+    nginx.ingress.kubernetes.io/proxy-buffer-size: "8k"
+    nginx.ingress.kubernetes.io/proxy-buffers: "4.0"
```
Comment (antgamdia) [inline script/chart-museum.sh:92]: not sure we really need these annotations here, no?

File: script/chart-museum.sh [hunk 3]
```diff
@@ -88,17 +79,47 @@ installChartMuseum() {
...
+      paths:
+      - backend:
+          service:
+            name: chartmuseum
+            port:
+              number: 8080
+        path: /
+        pathType: ImplementationSpecific
+EOF
+  sleep 20
```
Comment (antgamdia) [inline script/chart-museum.sh:108]: Why this manual sleep here? Maybe it's for giving some time to the ingress controller to become available? If so, why not just sth like: `        kubectl wait --namespace ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=120s
`

**Review by antgamdia — APPROVED** [review]
Great, thanks! Let's see if it brings more reliability to the CI runs 🤞

File: script/chart-museum.sh (hunk 3 above)
Comment (castelblanque) [reply script/chart-museum.sh:108]: That `sleep` is the key that missing. Before that, `Ingress` wasn't ready by the time it was used some instructions after. I thought about adding the `kubectl wait`, but that is for the nginx controller itself which is ready way before ([when the cluster is installed](https://github.com/vmware-tanzu/kubeapps/blob/bfec7d10a1d87e2b7808e3244bada03e7fa40a99/.circleci/config.yml#L364)).
It seems that it takes some time for the route to be processed by the Nginx controller?

File: script/chart-museum.sh (hunk 2 above)
Comment (castelblanque) [reply script/chart-museum.sh:92]: We are using those annotations in the other Ingress definition we use, so I just copy-pasted. Happy to remove them.

**Pull request merged**